File: app/src/main/java/com/letianpai/robot/desktop/callback/OpenAppCallback.java
package com.letianpai.robot.desktop.callback;


import java.util.ArrayList;

/**
 * Created by liujunbin
 */

public class OpenAppCallback {

    private ArrayList<OpenAppCommandListener> mOpenAppCommandListener = new ArrayList<>();

    private static class OpenAppCallbackHolder {
        private static OpenAppCallback instance = new OpenAppCallback();
    }

    private OpenAppCallback() {
    }

    public static OpenAppCallback getInstance() {
        return OpenAppCallbackHolder.instance;
    }

    public interface OpenAppCommandListener {
        void onResponseOpenApp(String command);
    }

    public void setOpenAppListener(OpenAppCommandListener listener) {
        this.mOpenAppCommandListener.add(listener);
    }

    public void openApp(String packageName) {
        for (int i = 0;i< mOpenAppCommandListener.size();i++){
            if (mOpenAppCommandListener.get(i) != null) {
                mOpenAppCommandListener.get(i).onResponseOpenApp(packageName);
            }
        }
    }

}

File: app/src/main/java/com/letianpai/robot/desktop/callback/OpenAppCallback.kt
package com.letianpai.robot.desktop.callback


/**
 * Created by liujunbin
 */
class OpenAppCallback private constructor() {
    private val mOpenAppCommandListener = ArrayList<OpenAppCommandListener?>()

    private object OpenAppCallbackHolder {
        val instance: OpenAppCallback = OpenAppCallback()
    }

    fun interface OpenAppCommandListener {
        fun onResponseOpenApp(command: String?)
    }

    fun setOpenAppListener(listener: OpenAppCommandListener?) {
        mOpenAppCommandListener.add(listener)
    }

    fun openApp(packageName: String?) {
        for (i in mOpenAppCommandListener.indices) {
            if (mOpenAppCommandListener[i] != null) {
                mOpenAppCommandListener[i]!!.onResponseOpenApp(packageName)
            }
        }
    }

    companion object {
        val instance: OpenAppCallback
            get() = OpenAppCallbackHolder.instance
    }
}












